controller_manager:
  ros__parameters:
    update_rate: 100 # Hz

    # joint_trajectory_controller은 내가 어디서 어디로 움직이라고 좌표를 지정하면 smooth한 trajectory를 만들어줌.
    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    gripper_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

arm_controller:
  ros__parameters:
    # URDF와 joint_name, command_interfaces, state_interfaces를 모두 동일하게
    joints:
      - panda_joint1
      - panda_joint2
      - panda_joint3
      - panda_joint4
      - panda_joint5
      - panda_joint6
      - panda_joint7

    command_interfaces:
      - position

    state_interfaces:
      - position

    # 로봇이 원하는 위치에 잘 도달했는지 센서 피드백을 확인하지 않는 open_loop로 진행
    open_loop_control: true
    # 목표 경로 데이터에 위치 값이 없다면 속도를 적분하여 위치를 계산 -> 경로 생성 알고리즘이 속도 프로파일만 생성하는 경우(Moveit) 유용
    allow_integration_in_goal_trajectories: true

gripper_controller:
  ros__parameters:
    joints:
      - panda_finger_joint1
      - panda_finger_joint2

    command_interfaces:
      - position

    state_interfaces:
      - position

    open_loop_control: true
    allow_integration_in_goal_trajectories: true